src/service-worker.ts

```diff
@@ -1,165 +1 @@
-  const latestCacheMainKey = latestCacheKeys?.find((url) => url.includes('/index.html'))
-  const latestCacheMainKeyResponse = latestCacheMainKey ? await latestCache.match(latestCacheMainKey) : null
-
-  const latestCacheOtherKeys = latestCacheKeys.filter((url) => url !== latestCacheMainKey) || []
-
-  const cachePromises = outdatedCacheNames.map((cacheName) => {
-    const getCacheDone = async () => {
-      const cache = await caches?.open(cacheName)
-      const cacheKeys = (await cache?.keys())?.map((c) => c.url) || []
-      const cacheMainKey = cacheKeys?.find((url) => url.includes('/index.html'))
-      if (cacheMainKey && latestCacheMainKeyResponse) {
-        await cache.put(cacheMainKey, latestCacheMainKeyResponse.clone())
-      }
-
-      return Promise.all(
-        latestCacheOtherKeys
-          .filter((key) => !cacheKeys.includes(key))
-          .map((url) => cache.add(url).catch((r) => console.error(r)))
-      )
-    }
-    return getCacheDone()
-  })
-
-  return Promise.all(cachePromises)
-}
-
-clientsClaim()
-
-// Precache the relevant assets generated by the build process.
-const manifest = self.__WB_MANIFEST.filter((entry) => {
-  const url = typeof entry === 'string' ? entry : entry.url
-  // If this is a language file, skip. They are compiled elsewhere.
-  if (url.endsWith('.po')) {
-    return false
-  }
-
-  // If this isn't a var woff2 font, skip. Modern browsers only need var fonts.
-  if (url.endsWith('.woff') || (url.endsWith('.woff2') && !url.includes('.var'))) {
-    return false
-  }
-
-  return true
-})
-precacheAndRoute(manifest)
-
-// Set up App Shell-style routing, so that navigation requests are fulfilled
-// immediately with a local index.html shell. See
-// https://developers.google.com/web/fundamentals/architecture/app-shell
-const fileExtensionRegexp = new RegExp('/[^/?]+\\.[^/]+$')
-
-type IndexRegisterParams = { request: Request; url: URL }
-registerRoute((params: IndexRegisterParams) => {
-  const { request, url } = params
-
-  // console.log('[worker] Index.html', request, url, params)
-  // return false
-  // If this isn't a DNS domain skip. IPFS gateways may not have domain
-  // separation, so they cannot use App Shell-style routing.
-  const hostName = url.hostname
-  if (!DNS_DOMAINS.includes(hostName) && !hostName.endsWith(DEV_DOMAINS_SUFFIX)) {
-    // console.log('[worker] FALSE. hostname', hostName)
-    return false
-  }
-
-  // If this isn't a navigation, skip.
-  if (request.mode !== 'navigate') {
-    // console.log('[worker] FALSE. No navigate', request.mode)
-    return false
-  }
-
-  // If this looks like a URL for a resource, skip.
-  if (url.pathname.match(fileExtensionRegexp)) {
-    // console.log('[worker] FALSE. If this looks like a URL for a resource, skip.', request.mode)
-    return false
-  }
-
-  // console.log('[worker] TRUE!')
-  return true
-}, createHandlerBoundToURL(process.env.PUBLIC_URL + '/index.html'))
+import './serviceWorker'
```

Comment:
Since this is a relative path but you created a custom serviceWorker folder, I'm not sure that this is the file you want to import instead of the custom one.

Reply:
But to add to this point, would it work if we just leave this file and the `serviceWorkerRegistration.ts`, same as uniswap and then also create these 2 files in `src/custom` and use our modified versions?

Reply:
This file is original, it has modifications before.
`serviceWorkerRegistration.ts` - added custom file